addon/components/models-table/cell-edit-toggle.js

```diff
@@ -0,0 +1,156 @@
+import Component from '@ember/component';
+import layout from '../../templates/components/models-table/cell-edit-toggle';
+import { get, computed } from '@ember/object';
+import { resolve } from 'rsvp';
+
+export default Component.extend({
+  layout,
+
+  record: null,
+
+  /**
+   * Bound from {{#crossLink "Components.ModelsTable/themeInstance:property"}}ModelsTable.themeInstance{{/crossLink}}
+   *
+   * @property themeInstance
+   * @type object
+   * @default null
+   */
+  themeInstance: null,
+
+  /**
+   * Closure action sent on Edit Button being clicked
+   *
+   * Action will send one parameter - object with fields:
+   *
+   * * `record` - The record to be edited
+   *
+   * @type function
+   * @property editRowAction
+   * @default null
+   * @return Must return a truthy value to allow the row to enter the Edit state. May return a Promise.
+   */
+  editRowAction: null,
+
+  /**
+   * Closure action sent on Save Button being clicked
+   *
+   * Action will send one parameter - object with fields:
+   *
+   * * `record` - The record to be saved
+   *
+   * @type function
+   * @property saveRowAction
+   * @default null
+   * @return Must return a truthy value to allow the row to exit the Edit state. May return a Promise.
+   */
+  saveRowAction: null,
+
+  /**
+   * Closure action sent on Cancel Button being clicked
+   *
+   * Action will send one parameter - object with fields:
+   *
+   * * `record` - The record currently being edited
+   *
+   * @type function
+   * @property cancelRowAction
+   * @default null
```

Comment:
Bound closure actions are marked as `@event` and not `@property` with `@type` and `default`